src/Misc/MoveToAbstractAlgebra.jl

```diff
@@ -0,0 +1,99 @@
+###############################################################################
+# A place to accumulate code that should eventually be moved to AbstractAlgebra.jl
+###############################################################################
+
+function symbols(Kt::Generic.RationalFunctionField)
+  return Kt.S
+end
+
+function number_of_generators(F::AbstractAlgebra.Generic.FracField{T}) where {T <: MPolyRingElem}
+  return number_of_generators(base_ring(F))
+end
+
+function gen(F::AbstractAlgebra.Generic.FracField{T}) where {T <: PolyRingElem}
+  return F(gen(base_ring(F)))
+end
+
+function gen(F::AbstractAlgebra.Generic.FracField{T}, i::Int) where {T <: MPolyRingElem}
+  return F(gen(base_ring(F), i))
+end
+
+function gens(F::AbstractAlgebra.Generic.FracField{T}) where {T <: Union{PolyRingElem, MPolyRingElem}}
+  return map(F, gens(base_ring(F)))
+end
+
+"""
+    is_alternating(B::MatElem)
+
+Return whether the form corresponding to the matrix `B` is alternating,
+i.e. `B = -transpose(B)` and `B` has zeros on the diagonal.
+Return `false` if `B` is not a square matrix.
+"""
+function is_alternating(B::MatElem)
+   n = nrows(B)
+   n==ncols(B) || return false
+
+   for i in 1:n
+      B[i,i]==0 || return false
+      for j in i+1:n
+         B[i,j]==-B[j,i] || return false
+      end
+   end
+
+   return true
+end
+
+@doc raw"""
+    upper_triangular_matrix(L)
+
+Return the upper triangular matrix whose entries on and above the diagonal are the elements of `L`.
+
+An exception is thrown whenever the length of `L` is not equal to $n(n+1)/2$,
+for some integer $n$.
+"""
+function upper_triangular_matrix(L)
```

Comment:
I know this PR just moves things around, but the code as it is definitely is not ready for broad consumption...


That said, it seems AA actually already defines it (and also `lower_triangular_matrix`) but with a much better docstring and a slightly nicer implementation.

So perhaps we can just delete it here?